### ISSUE TITLE
Add preliminary Spring Batch support

### ISSUE THREAD

**Issue by dsyer** [spring-projects-experimental/spring-native#203]
The Batch samples look like they are using generated JSON. It would be nice to tighten that up.

**Comment by aclement** [comment]
Yes, I don't think it was pulled back into hints, I was just relieved when it was working. We should do that.

**Comment by sdeleuze** [comment]
@aclement I am not sure we can properly support Batch in 0.8 yet, I have tried to polish a little bit the build config and I get this error message:
```
org.springframework.beans.factory.BeanDefinitionStoreException: @Configuration classes need to be marked as proxyBeanMethods=false. Found: [org.springframework.batch.core.configuration.annotation.ScopeConfiguration, org.springframework.batch.core.configuration.annotation.SimpleBatchConfiguration]
```

@mminella Could you please update Batch to use `proxyBeanMethods=false`?

**Comment by mminella** [comment]
Depends on https://github.com/spring-projects-experimental/spring-native/issues/459

**Comment by sdeleuze** [comment]
@mminella @aclement FYI batch is now using the build-time infrastructure like other samples, and to make it work I had to specify [this native hint](https://github.com/spring-projects-experimental/spring-native/blob/master/spring-native-samples/batch/src/main/java/com/example/batch/BatchApplication.java#L21).

Since 0.9.0 is getting closer and #459 does not look like a trivial change, I would like to propose to postpone both #459 and batch support to 0.9.1 is order to allow us to focus our time and energy on polishing bti on what is currently supported, and to mature properly batch support before announcing it. Any thoughts?

**Comment by mminella** [comment]
@sdeleuze I have no objections as batch support is currently blocked on #459.

**Comment by mminella** [comment]
@sdeleuze Why would that hint be added there and not to `BatchHints`?

**Comment by sdeleuze** [comment]
@mminella It was temporary since I was not sure if it was Batch specific or not, but I can indeed move it to `BatchHints`.

After discussion with @aclement, we agreed that #459 and more generally batch support will require more work, so it is more reasonsable to move it to 0.9.1.

**Comment by sdeleuze** [comment]
As discussed in https://github.com/spring-projects-experimental/spring-native/pull/667#discussion_r603449707, among other things, [the batch sample `build.sh`](https://github.com/spring-projects-experimental/spring-native/blob/master/samples/batch/build.sh) should be updated to avoid passing custom arguments in order to ensure the build is easily reproducible.

**Comment by snicoll** [comment]
@sdeleuze looking at the history of this issue, I am not sure what is remaining.

**Comment by sdeleuze** [comment]
`batch` sample is fine, `batch-io` is not since [it is still using JSON configuration file](https://github.com/spring-projects-experimental/spring-native/tree/main/samples/batch-io/src/main/resources/META-INF/native-image). I will try to turn that to general batch hints, and add application specific hints to the application class which already contains an `@AotProxyHint` [here](https://github.com/spring-projects-experimental/spring-native/blob/main/samples/batch-io/src/main/java/com/example/batch/BatchApplication.java#L8).

That should allow us to claim preliminary support.
